src/needless_bool.rs

```diff
@@ -78,6 +93,65 @@ impl LateLintPass for NeedlessBool {
     }
 }
 
+#[derive(Copy,Clone)]
+pub struct BoolComparison;
+
+impl LintPass for BoolComparison {
+    fn get_lints(&self) -> LintArray {
+        lint_array!(BOOL_COMPARISON)
+    }
+}
+
+impl LateLintPass for BoolComparison {
+    fn check_expr(&mut self, cx: &LateContext, e: &Expr) {
+        if let ExprBinary(Spanned{ node: BiEq, .. }, ref left_side, ref right_side) = e.node {
+            match (fetch_bool_expr(left_side), fetch_bool_expr(right_side)) {
+                (Some(true), None) => {
+                    let hint = format!("{}", snippet(cx, right_side.span, ".."));
```

Comment:
You never need `format!("{}", foo)`¹. Here `foo.into()` should do it.

1: We should have a lint for that.